src/OpenLoco/Windows/Construction/ConstructionTab.cpp

```diff
@@ -1831,7 +1830,7 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
 
         if (_constructionHover == 1)
         {
-            if (!Input::isToolActive(WindowType::construction, self->number) || _toolWidgetIndex != widx::construct)
+            if (!Input::isToolActive(WindowType::construction, self->number) || WindowManager::getToolWidgetIndex() != widx::construct)
```

Comment:
there is a different overload available that this can use `Input::isToolActive(WindowType::construction, self->number, widx::construct)`